native/src/TableclothSet.mli

```diff
@@ -0,0 +1,339 @@
+(** A {!Set} represents a unique collection of values.
+
+    [Set] is an immutable data structure which means operations like {!Set.add} and {!Set.remove} do not modify the data structure, but return a new set with the desired changes.
+
+    Since the usage is so common the {!Set.Int} and {!Set.String} modules are available, offering a convenient way to construct sets.
+
+    For other data types you can use {!Set.Poly} which uses OCaml's polymorphic [compare] function.
+*)
```

Comment:
A comment showing us how to create a Set of a different type would be super useful.